### ISSUE TITLE
grains.list_present produces incorrect (?) output

### ISSUE THREAD

**Issue by kkaig** [saltstack/salt#23548]
Assuming an empty grains file on the minion, and a state file on the master that contains the following:
```yaml
roles:
  grains.list_present:
    - value:
      - item1
      - item2
      - item3
```

I would expect it to produce the following output in the minion grain:
```yaml
roles:
  - item1
  - item2
  - item3
```

But instead it produces the following:
```yaml
roles:
- - item1
  - item2
  - item3
```

This has been confirmed by others on IRC. Salt version is 2014.7.5. Feel free to delete this if it's not a bug, or edit it in accordance with guidelines (it's my first report, and it's been a long day...)

Thanks

**Comment by sdm24** [comment]
I can confirm this bug, also on salt 2014.7.5

When I change the state to this:
```Saltstack
roles1:
  grains.list_present:
    - name: roles
    - value: item1
roles2:
  grains.list_present:
    - name: roles
    - value: item2
roles3:
  grains.list_present:
    - name: roles
    - value:
      - item3
      - item4
```
The result I get is:
```Saltstack
    roles:
        - item1
        - item2
        |_
          - item3
          - item4
```
So it seems that the issue is specifically when trying to make a list, using the format found at http://docs.saltstack.com/en/latest/ref/states/all/salt.states.grains.html#salt.states.grains.list_present

**Comment by jfindlay** [comment]
@kkaig, @sdm24, thanks for the report.

**Comment by cachedout** [comment]
OK @sdm24 and @kkaig, give the diff in #23674 a try and see if it works better. Thanks!

**Comment by kkaig** [comment]
Sorry for being dumb, but I'm new to git and github and can't figure out how to to apply the diff. I've cloned the salt repo, but I don't want to copy paste the diff direct from github, and I can't find an option to download it raw.

Any docs on this? All I can find is stackoverflow arguments

**Comment by cachedout** [comment]
Hi @kkaig. You could clone my PR branch that includes the fix:

https://github.com/cachedout/salt/tree/issue_23674

Or if you want the raw diff, it is here: https://patch-diff.githubusercontent.com/raw/saltstack/salt/pull/23674.diff

**Comment by kkaig** [comment]
Really sorry Mike...I've tried cloning your branch and I can't see the change. I tried applying the diff and it wouldn't apply. I tried downloading the zip directly and still couldn't see it. Like I said, I really don't know anything about git, so I'm just randomly copy pasting commands, which is not a sensible nor productive way to work. You'll have to wait for someone else who understands git enough to confirm the bug's fixed (...or wait a few days while I go through the git tutorials and start to understand what's going on here)

Again, sorry

Kyle

**Comment by sdm24** [comment]
Sorry, I won't be back at my work computer until Monday. I will test the code then

**Comment by cachedout** [comment]
Don't be sorry. :] We're all learning new things.

You probably just need to checkout the specific branch after you clone the repo. Something like `.git checkout issue_23674`. Let me know if I can help.

**Comment by L4rS6** [comment]
I got the same problem in Version 2015.5.0+ds-1trusty1:

init.sls:
roles:
  grains.list_present:
    - value: ['test','test2']

/etc/salt/grains:
roles:
- - test
  - test2

Has that fix not been added to the new version?

**Comment by jfindlay** [comment]
@L4rS6, the fix came in just after 2015.5.0 was tagged.  It is in 2015.5.1 and later.
